### PR TITLE
IvorySQL: compatible oracle sequence

### PR DESCRIPTION
**Pull request by tanyang-star**

### CODE REVIEW

**Comment by tanyang-star** [comment]
Add: compatible oracle sequence #228

File: src/backend/catalog/information_schema.sql [hunk 1]
```diff
@@ -758,8 +758,8 @@ CREATE VIEW columns AS
            CAST(a.attnum AS sql_identifier) AS dtd_identifier,
            CAST('NO' AS yes_or_no) AS is_self_referencing,
 
-           CAST(CASE WHEN a.attidentity IN ('a', 'd') THEN 'YES' ELSE 'NO' END AS yes_or_no) AS is_identity,
-           CAST(CASE a.attidentity WHEN 'a' THEN 'ALWAYS' WHEN 'd' THEN 'BY DEFAULT' END AS character_data) AS identity_generation,
+           CAST(CASE WHEN a.attidentity IN ('a', 'd', 'n', 'i', 'o') THEN 'YES' ELSE 'NO' END AS yes_or_no) AS is_identity, /* IvorySQL: sequence */
```
Comment (wangjie-star) [inline src/backend/catalog/information_schema.sql:761]: sql-sequence?

File: src/backend/commands/sequence.c [hunk 2]
```diff
@@ -1365,6 +1366,19 @@ init_params(ParseState *pstate, List *options, bool for_identity,
 					 errmsg("invalid sequence option SEQUENCE NAME"),
 					 parser_errposition(pstate, defel->location)));
 		}
+		/* IvorySQL:BEGIN - sql-sequence */
+		else if(strcmp(defel->defname, "order") == 0 || strcmp(defel->defname, "noorder") == 0)
```
Comment (wangjie-star) [inline src/backend/commands/sequence.c:1370]: Space is required after “else if”

File: src/backend/commands/sequence.c [hunk 3]
```diff
@@ -1365,6 +1366,19 @@ init_params(ParseState *pstate, List *options, bool for_identity,
...
+		{
+			ordercnt++;
+			if(ordercnt > 1)
```
Comment (wangjie-star) [inline src/backend/commands/sequence.c:1373]: Space is required after “if”

File: src/backend/commands/sequence.c [hunk 4]
```diff
@@ -1382,14 +1396,35 @@ init_params(ParseState *pstate, List *options, bool for_identity,
...
+			if (newtypid != INT2OID &&
+				newtypid != INT4OID &&
+				newtypid != INT8OID)
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 for_identity
+						 ? errmsg("identity column type must be smallint, integer, or bigint")
+						 : errmsg("sequence type must be smallint, integer, or bigint")));
+		}
+		else if(seq_type == ATTRIBUTE_IDENTITY_DEFAULT_ON_NULL || seq_type == ATTRIBUTE_ORA_IDENTITY_ALWAYS
```
Comment (wangjie-star) [inline src/backend/commands/sequence.c:1411]: Space is required after “else if”

File: src/backend/commands/tablecmds.c [hunk 5]
```diff
@@ -7981,6 +7981,11 @@ ATExecDropIdentity(Relation rel, const char *colName, bool missing_ok, LOCKMODE
 		}
 	}
 
+	/* IvorySQL:BEGIN - sql-sequence */
+	if(attTup->attidentity == ATTRIBUTE_IDENTITY_DEFAULT_ON_NULL)
```
Comment (wangjie-star) [inline src/backend/commands/tablecmds.c:7985]: Space is required after “if”

File: src/backend/oracle_parser/ora_gram.y [hunk 6]
```diff
@@ -3920,10 +3975,28 @@ ColConstraintElem:
...
+					if($5)
+					{
+						str = scanner_querytext(@5, @5+1, yyscanner);
+					}
+					/* IvorySQL:END - sql-sequence */
+
 					n->contype = CONSTR_IDENTITY;
 					n->generated_when = $2;
+					/* IvorySQL:BEGIN - sql-sequence */
+					if( $5 != NULL && strncmp(str,"(",1) != 0 && n->generated_when  == ATTRIBUTE_IDENTITY_ALWAYS)
```
Comment (wangjie-star) [inline src/backend/oracle_parser/ora_gram.y:3991]: Space is required after “if”

**Pull request merged**